spec/controllers/generic_files_controller_spec.rb

```diff
@@ -0,0 +1,40 @@
+require 'spec_helper'
+
+describe GenericFilesController, type: :controller do
+  routes { Sufia::Engine.routes }
+  let(:user) { FactoryGirl.create(:user) }
+  before do
+    allow(controller).to receive(:has_access?).and_return(true)
+    sign_in user
+    allow_any_instance_of(User).to receive(:groups).and_return([])
+    allow(controller).to receive(:clear_session_user) ## Don't clear out the authenticated session
+    allow_any_instance_of(GenericFile).to receive(:characterize)
+  end
+
+  describe "PUT #update" do
+    let(:generic_file) do
+      GenericFile.create do |gf|
+        gf.apply_depositor_metadata(user)
+        gf.title = ['test title']
+      end
+    end
+
+    context "when updating metadata" do
+      let(:update_message) { double('content update message') }
+      before do
+        allow(ContentUpdateEventJob).to receive(:new).with(generic_file.id, 'test_user1@example.com').and_return(update_message)
```

Comment:
Ok. The test should make sure that when the edit page is posted, it redirects to the show page IF the metadata is valid, and back to the edit page (with a flash) if the metadata is not.